canonical/ubuntu-advantage-client
# Rewrite unit test environments for the different releases (SC-1044)

Define `tox` environments to run the tests with different python versions, and add a script to set up `pyenv`.
Make sure all unit tests and flake8/mypy checks are executed with respect to the Python version, setuptools version, and package version constraints for the supported releases.

I tested the pyenv solution on Focal, Impish, Jammy, and it worked.

## Test Steps
`make clean`
~`make testdeps`~ run the `tools/setup_pyenv.sh` script
Then make sure pyenv is working by refreshing your terminal, and then
`tox`
Everything should run and pass.

## Desired commit type

## Checklist
